examples/applications/plot_time_series_lagged_features.py

```diff
@@ -258,36 +268,33 @@ def consolidate_scores(cv_results, scores, metric):
             metric = key.split("test_")[1]
             scores = consolidate_scores(cv_results, scores, metric)
 
-df = pd.DataFrame(scores)
-
-styled_df_copy = df.copy()
-
+styled_df = pl.DataFrame(scores)
 
-def extract_numeric(value):
-    parts = value.split("±")
-    mean_value = float(parts[0])
-    std_value = float(parts[1].split()[0])
 
-    return mean_value, std_value
+def extract_numeric(col):
+    col_split = pl.col(col).str.split(by=" ")
+    return pl.concat_list([col_split.list.get(0), col_split.list.get(2)]).cast(
+        pl.List(pl.Float64)
+    )
 
 
-# Convert columns containing "±" to tuples of numerical values
-cols_to_convert = df.columns[1:]  # Exclude the "loss" column
-for col in cols_to_convert:
-    df[col] = df[col].apply(extract_numeric)
+def list_min(col):
+    # Polars doesn't currently support finding a minimum of a list
+    # so we find the minimum by sorting.
+    return col.sort_by([col.list.first(), col.list.last()]).first()
 
-min_values = df.min()
 
-# Create a mask for highlighting minimum values
-mask = pd.DataFrame("", index=df.index, columns=df.columns)
-for col in cols_to_convert:
-    mask[col] = df[col].apply(
-        lambda x: "font-weight: bold" if x == min_values[col] else ""
+gt_styled_df = GT(styled_df)

```

Comment:
For sure, we don't want the extra dependency on `great-tables`. We should instead have a way to do the same with the current installed library.

Reply:
It seems that there is no way to print something in bold in Polars, that's why I added great-tables that have this support. 

Should I then convert the dataframe back to pandas when there is a styled output? As I thought you don't want to have both pandas and Polars dataframes https://github.com/scikit-learn/scikit-learn/pull/28601#issuecomment-1988098756? 

Or do you know about some other way to print that table with final metrics with bold styling?

Alternatively, should I display the dataframe without styling? I can add the minimums as a separate dataframe below.

Reply:
I guess we rather lose the bold styling than adding the extra dependency or jumping back and forth from pandas. Personally I vote for the last option then.

Reply:
Thanks for your reviews. I removed the styling with great tables and added a dataframe displaying the losses that minimise each metric.